src/PlusImageProcessing/vtkPlusTransverseProcessEnhancer.cxx

```diff
@@ -24,125 +24,100 @@ See License.txt for details.
-
-    Thresholder(vtkSmartPointer<vtkImageThreshold>::New()),
-    GaussianEnabled(false),
-    ThresholdInValue(0.0),
-    ThresholdOutValue(255.0),
-    LowerThreshold(0.0),
-    UpperThreshold(0.0),
-
-    GaussianSmooth(vtkSmartPointer<vtkImageGaussianSmooth>::New()),
-    EdgeDetector(vtkSmartPointer<vtkImageSobel2D>::New()),
-    ImageBinarizer(vtkSmartPointer<vtkImageThreshold>::New()),
-    BinaryImageForMorphology(vtkSmartPointer<vtkImageData>::New()),
-    IslandRemover(vtkSmartPointer<vtkImageIslandRemoval2D>::New()),
-    ImageEroder(vtkSmartPointer<vtkImageDilateErode3D>::New()),
-
-    ReturnToFanImage(false),
-    CurrentFrameMean(0.0),
-    CurrentFrameStDev(0.0),
-    CurrentFrameMax(0.0),
-    CurrentFrameMin(255.0),
-    
-    ThresholdingEnabled(false),
-    
-    EdgeDetectorEnabled(false),
-    ConversionImage(vtkSmartPointer<vtkImageData>::New()),
-    IslandRemovalEnabled(false),
-    IslandAreaThreshold(-1),
-    ErosionEnabled(false),
-    DilationEnabled(false),
-    ReconvertBinaryToGreyscale(false),
-    
-    LinesImage(vtkSmartPointer<vtkImageData>::New()),
-    //LinesFrame(),
-    //LinesImageList(vtkSmartPointer<vtkPlusTrackedFrameList>::New()),
-    LinesFrameList(vtkPlusTrackedFrameList::New()),
-    UnprocessedLinesImage(vtkSmartPointer<vtkImageData>::New()),
-    ShadowImage(vtkSmartPointer<vtkImageData>::New()),
-    //ShadowFrame(),
-    //ShadowImageList(vtkSmartPointer<vtkPlusTrackedFrameList>::New()),
-    ShadowFrameList(vtkPlusTrackedFrameList::New()),
-    IntermediateImage(vtkSmartPointer<vtkImageData>::New()),
-    //IntermediateFrame(),
-    //IntermediateImageList(vtkSmartPointer<vtkPlusTrackedFrameList>::New()),
-    IntermediateFrameList(vtkPlusTrackedFrameList::New()),
-    ProcessedLinesImage(vtkSmartPointer<vtkImageData>::New()),
-    //ProcessedLinesFrame(),
-    //ProcessedLinesImageList(vtkSmartPointer<vtkPlusTrackedFrameList>::New())
-    ProcessedLinesFrameList(vtkPlusTrackedFrameList::New())
+  : ScanConverter(NULL),
+  NumberOfScanLines(0),
+  NumberOfSamplesPerScanLine(0),
+
+  RadiusStartMm(0),
+  RadiusStopMm(0),
+  ThetaStartDeg(0),
+  ThetaStopDeg(0),
+
+  GaussianSmooth(NULL),
+  EdgeDetector(NULL),
+  ImageBinarizer(NULL),
+  BinaryImageForMorphology(NULL),
+  IslandRemover(NULL),
+  ImageEroder(NULL),
+  ImageDialator(NULL),
+
+  ConversionImage(NULL),
+  IslandAreaThreshold(-1),
+  BoneOutlineDepthPx(3), //Note: this only changes the apperance/thickness of the 3D model. Different numbers do not change what is or isnt marked as bone.
+  BonePushBackPx(9),     //Horisontal distance between where a shadow is located, and where the bone begins
+
+  LinesImage(NULL),
+  ProcessedLinesImage(NULL),
+  UnprocessedLinesImage(NULL)
+
 {
-  this->SetDilationKernelSize(0, 0);
+  this->SetMmToPixelFanImage(0, 0, 0);
+  this->SetMmToPixelLinesImage(0, 0, 0);
+
+  this->GaussianSmooth = vtkSmartPointer<vtkImageGaussianSmooth>::New();    //Used to smooth the image

```

Comment:
Please return these to the constructor list

Reply:
Yes, see, here. I see, he's getting two different instructions.

So, my original question, is this a style or performance reason?

Reply:
Style, but I guess for smart pointers where deleting them is not a concern they are fine in the constructor list.

Reply:
Ok, I don't mind having them in the body. I'll switch.

Reply:
In recent nightly versions, vtkNew<> does not require .GetPointer() anymore to get the pointer, so we can use that most places instead of vtkSmartPointer<>.
vtkNew<> works in the header, so no additional code will be needed in the constructor.